spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPartitioningAwareScan.java

```diff
@@ -78,7 +78,19 @@ abstract class SparkPartitioningAwareScan<T extends PartitionScanTask> extends S
       Schema expectedSchema,
       List<Expression> filters,
       Supplier<ScanReport> scanReportSupplier) {
-    super(spark, table, readConf, expectedSchema, filters, scanReportSupplier);
+    this(spark, table, scan, readConf, expectedSchema, filters, scanReportSupplier, null);
+  }
+
+  SparkPartitioningAwareScan(
+      SparkSession spark,
+      Table table,
+      Scan<?, ? extends ScanTask, ? extends ScanTaskGroup<?>> scan,
+      SparkReadConf readConf,
+      Schema expectedSchema,
+      List<Expression> filters,
+      Supplier<ScanReport> scanReportSupplier,
+      Integer pushedLimit) {
```

Comment:
I'm wondering if we should include this in SparkReadConf rather than having a separate argument. I have similar thoughts around SparkInputPartition. I'm not a big fan of having to plumb the new arguments all the way through the code base but those two options may not look great either since they aren't a great fit imho. 

Ideally I think I would want something like SparkReadContext but I don't know how often more things like this will come up

Reply:
Thanks @RussellSpitzer for sharing your concern and suggestion!
I think we can borrow @aokolnychyi 's idea of adding [`ParquetBatchReadConf` and `OrcBatchReadConf`](https://github.com/apache/iceberg/pull/9841#discussion_r1594829408). We can have something like 
```
@Value.Immutable
public interface ParquetBatchReadConf extends Serializable {
  ParquetReaderType readerType();  // this is for comet, we don't need this for now
  int batchSize();
  @Nullable
  Integer limit();
}

@Value.Immutable
public interface OrcBatchReadConf extends Serializable {
  int batchSize();
}
```
Similarly, we can also have `ParquetRowReadConf` and `OrcRowReadConf`.

I have changed the code to add `ParquetBatchReadConf` and `OrcBatchReadConf`. We still need to pass `pushedLimit` to the `SparkPartitioningAwareScan`, `SparkScan` and `SparkBatch` constructors, so `pushedLimit` can be passed from `SparkScanBuilder` to `SparkBatch`, this is because `pushedLimit` is not available in `SparkReadConf`, we have to call `SparkScanBuilder`.`pushLimit` to get `pushedLimit`.

Please let me know if this approach looks OK to you. Thanks!